cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1225,6 +1226,85 @@ public void getInstances(FullHttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(output));
   }
 
+  /**
+   * Returns the run counts for all program runnables that are passed into the data. The data is an array of
+   * Json objects where each object must contain the following three elements: appId, programType, and programId.
+   * <p>
+   * Example input:
+   * <pre><code>
+   * [{"appId": "App1", "programType": "Service", "programId": "Service1"},
+   *  {"appId": "App1", "programType": "Workflow", "programId": "testWorkflow"},
+   *  {"appId": "App2", "programType": "Workflow", "programId": "DataPipelineWorkflow"}]
+   * </code></pre>
+   * </p><p>
+   * </p><p>
+   * The response will be an array of JsonObjects each of which will contain the three input parameters
+   * as well as 2 fields, "count" which maps to the count of the program and "statusCode" which maps to the
+   * status code for the data in that JsonObjects.
+   * </p><p>
+   * If an error occurs in the input (for the example above, workflow in app1 does not exist),
+   * then all JsonObjects for which the parameters have a valid status will have the count field but all JsonObjects
+   * for which the parameters do not have a valid status will have an error message and statusCode.
+   * </p><p>
+   * For example, if there is no workflow in App1 in the data above, then the response would be 200 OK with following
+   * possible data:
+   * </p>
+   * <pre><code>
+   * [{"appId": "App1", "programType": "Service", "programId": "Service1",
+   * "statusCode": 200, "count": 20},
+   * {"appId": "App1", "programType": "Workflow", "programId": "testWorkflow", "statusCode": 404,
+   * "error": "Program 'testWorkflow' is not found"},
+   *  {"appId": "App2", "programType": "Workflow", "programId": "DataPipelineWorkflow", "runnableId": "Flowlet1",
+   *  "statusCode": 200, "count": 300}]
+   * </code></pre>
+   */
+  @POST
+  @Path("/runcount")
+  public void getRuncounts(FullHttpRequest request, HttpResponder responder,
+                           @PathParam("namespace-id") String namespaceId) throws Exception {
+    List<BatchProgram> programs = validateAndGetBatchInput(request, BATCH_PROGRAMS_TYPE);
+
+    List<BatchProgramCount> counts = new ArrayList<>(programs.size());
+    for (BatchProgram program : programs) {
+      ProgramId programId = new ProgramId(namespaceId, program.getAppId(), program.getProgramType(),
+                                          program.getProgramId());
+      try {
+        int count = lifecycleService.getProgramRuncount(programId);
+        counts.add(new BatchProgramCount(program, HttpResponseStatus.OK.code(), null, count));
+      } catch (NotFoundException e) {
+        counts.add(new BatchProgramCount(program, HttpResponseStatus.NOT_FOUND.code(), e.getMessage(), -1));
```

Comment:
count value -1 means the result json will have this number. Either update the javadoc or use Integer type for count in `BatchProgramCount`

Reply:
better readability:

```java
int count = -1;
HttpResponseStatus status = null;
String errorMessage = null;
try {
        count = lifecycleService.getProgramRuncount(programId);
        status = HttpResponseStatus.OK;
      } catch (NotFoundException e) {
        status = HttpResponseStatus.NOT_FOUND;
         errorMessage = e.getMessage();
      } catch (UnauthorizedException e) {
         status = HttpResponseStatus.FORBIDDEN;
         errorMessage = e.getMessage();
      }
counts.add(new BatchProgramCount(program, status.code(), errorMessage, count)
```